tools/stern/src/Stern/Types.hs

```diff
@@ -0,0 +1,90 @@
+{-# LANGUAGE DataKinds                  #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE NamedFieldPuns             #-}
+{-# LANGUAGE OverloadedStrings          #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE StandaloneDeriving         #-}
+{-# LANGUAGE TemplateHaskell            #-}
+
+module Stern.Types where
+
+import Brig.Types
+import Data.Aeson
+import Data.Aeson.TH
+import Data.ByteString.Conversion
+import Data.Json.Util
+import Data.Range
+import Data.Text
+import Imports
+import Galley.Types.Teams
+import Galley.Types.Teams.Intra
+
+import qualified Data.HashMap.Strict as M
+
+newtype TeamMemberInfo = TeamMemberInfo { tm :: TeamMember }
+
+instance ToJSON TeamMemberInfo where
+    toJSON (TeamMemberInfo m) =
+        let Object o = teamMemberJson (const True) m
+        in Object $ M.insert "can_update_billing" (Bool (hasPermission m SetBilling))
+                  $ M.insert "can_view_billing"   (Bool (hasPermission m GetBilling))
+                  $ o
+
+data TeamInfo = TeamInfo
+    { tiData    :: TeamData
+    , tiMembers :: [TeamMemberInfo]
+    }
+
+instance ToJSON TeamInfo where
+    toJSON (TeamInfo d m) = object
+        [ "info"    .= d
+        , "members" .= m
+        ]
+
+newtype InvoiceId = InvoiceId { unInvoiceId :: Text }
+    deriving (Eq, Show, ToByteString, FromByteString, ToJSON, FromJSON)
+
+data TeamBillingInfo = TeamBillingInfo
```

Comment:
This refers to information of components that are not (yet?) open-sourced. Some comments explaining this might be useful. 

In addition, some endpoints will appear broken - perhaps a config flag can be used to enable or disable optional services like galeb/ibis?